src/lib/FlightTasks/tasks/FlightTaskAuto.hpp

```diff
@@ -84,13 +92,28 @@ class FlightTaskAuto : public FlightTask
 	WaypointType _type{WaypointType::idle}; /**< Type of current target triplet. */
 	uORB::Subscription<home_position_s> *_sub_home_position{nullptr};
 
+
+	State _current_state{State::none};
+
+	float _speed_at_target = 0.0f; /**< Desired velocity at target. */
+
+	DEFINE_PARAMETERS_CUSTOM_PARENT(FlightTask,
+					(ParamFloat<px4::params::MPC_XY_CRUISE>) MPC_XY_CRUISE,
+					(ParamFloat<px4::params::MPC_CRUISE_90>) MPC_CRUISE_90, // speed at corner when angle is 90 degrees move to line
+					(ParamFloat<px4::params::NAV_ACC_RAD>) NAV_ACC_RAD // acceptance radius at which waypoints are updated move to line
+				       ); /**< Default mc cruise speed.*/
+
+
 private:
 	matrix::Vector2f _lock_position_xy{NAN, NAN};
 
 	uORB::Subscription<position_setpoint_triplet_s> *_sub_triplet_setpoint{nullptr};
 
-	DEFINE_PARAMETERS_CUSTOM_PARENT(FlightTask,
-					(ParamFloat<px4::params::MPC_XY_CRUISE>) MPC_XY_CRUISE); /**< Default mc cruise speed.*/
+	matrix::Vector3f _triplet_target;
+	matrix::Vector3f _triplet_prev_wp;
+	matrix::Vector3f _triplet_next_wp;
+	matrix::Vector2f _closest_pt;
```

Comment:
add description: closest point to what? 
Is this required to be a class member?

Reply:
yes, you then need to use it as the target it the state is offtrack.